src/Type.php

```diff
@@ -469,8 +469,10 @@ public function save(EntityInterface $entity, $options = [])
         $type = $this->connection()->getIndex()->getType($this->name());
         $id = $entity->id ?: null;
 
-        $data = $entity->toArray();
-        unset($data[$id]);
+        $keys = array_keys($entity->toArray());
+        $metaKeys = ['id', '_version'];
+        $data = $entity->extract(array_diff($keys, $metaKeys));
```

Comment:
Yeah just thought this was more future proofed.

Also it would need to be `unset($data['id'], $data['_version']`